extensions/ql-vscode/src/local-queries.ts

```diff
@@ -7,384 +7,575 @@ import {
-              ? `${queriesRemaining} remaining. ${update.message}`
-              : update.message;
-          progress({
-            ...update,
-            message,
+
+          wrappedProgress({
+            maxStep: queryUris.length,
+            step: queryUris.length - queriesRemaining,
+            message: "",
           });
-        }
 
-        wrappedProgress({
-          maxStep: queryUris.length,
-          step: queryUris.length - queriesRemaining,
-          message: "",
-        });
-
-        await Promise.all(
-          queryUris.map(async (uri) =>
-            compileAndRunQuery(
-              queryRunner,
-              queryHistoryManager,
-              databaseUI,
-              localQueryResultsView,
-              queryStorageDir,
-              false,
-              uri,
-              wrappedProgress,
-              token,
-              undefined,
-            ).then(() => queriesRemaining--),
+          await Promise.all(
+            queryUris.map(async (uri) =>
+              this.compileAndRunQuery(
+                false,
+                uri,
+                wrappedProgress,
+                token,
+                undefined,
+              ).then(() => queriesRemaining--),
+            ),
+          );
+        },
+        {
+          title: "Running queries",
+          cancellable: true,
+        },
+      );
+
+    const quickEval = async (uri: Uri) =>
+      withProgress(
+        async (progress, token) => {
+          await this.compileAndRunQuery(true, uri, progress, token, undefined);
+        },
+        {
+          title: "Running query",
+          cancellable: true,
+        },
+      );
+
+    const codeLensQuickEval = async (uri: Uri, range: Range) =>
+      withProgress(
+        async (progress, token) =>
+          await this.compileAndRunQuery(
+            true,
+            uri,
+            progress,
+            token,
+            undefined,
+            range,
           ),
-        );
-      },
-      {
-        title: "Running queries",
-        cancellable: true,
-      },
-    );
+        {
+          title: "Running query",
+          cancellable: true,
+        },
+      );
 
-  const quickEval = async (uri: Uri) =>
-    withProgress(
-      async (progress, token) => {
-        await compileAndRunQuery(
-          queryRunner,
-          queryHistoryManager,
-          databaseUI,
-          localQueryResultsView,
-          queryStorageDir,
-          true,
-          uri,
-          progress,
-          token,
-          undefined,
-        );
-      },
-      {
-        title: "Running query",
-        cancellable: true,
-      },
-    );
+    const quickQuery = async () =>
+      withProgress(
+        async (progress, token) =>
+          displayQuickQuery(
+            this.app,
+            this.cliServer,
+            this.databaseUI,
+            progress,
+            token,
+          ),
+        {
+          title: "Run Quick Query",
```

Comment:
Do we want this?

```suggestion
          title: "Run Quick Query",
          cancellable: true,
```

Reply:
My first reponse was "yes, obvious oversight". However, looking at the implementation, the command seems to just show the Quick Query editor, rather than actually running the query. That doesn't seem like we need a "cancel" button.